clients/src/main/java/org/apache/kafka/clients/consumer/MockConsumer.java

```diff
@@ -424,12 +447,20 @@ private void resetOffsetPosition(TopicPartition tp) {
             if (offset == null)
                 throw new IllegalStateException("MockConsumer didn't have beginning offset specified, but tried to seek to beginning");
         } else if (strategy == OffsetResetStrategy.LATEST) {
-            offset = endOffsets.get(tp);
+            offset = getEndOffset(endOffsets.get(tp));
             if (offset == null)
                 throw new IllegalStateException("MockConsumer didn't have end offset specified, but tried to seek to end");
         } else {
             throw new NoOffsetForPartitionException(tp);
         }
         seek(tp, offset);
     }
+
+    private Long getEndOffset(List<Long> offsets) {
+
+        if(offsets == null || offsets.isEmpty()) {
```

Comment:
nit: missing space after `if` -- wondering why checkstyle does not complain?

Reply:
my fault, `MockConsumer` is in clients, I'm used to running `streams:clean streams:test` neglected to do so for `clients.